src/__tests__/index.test.ts

```diff
@@ -0,0 +1,512 @@
+    let testError: Error;
+    beforeEach(function () {
+      testError = new Error('Failed to release Pool');
+      createPoolStub.resolves(connectionPool);
+    });
+    context('When there is no connectionPool', function () {
+      beforeEach(async function () {
+        return wrapper.releasePool();
+      });
+      it('Should not call connectionPool.close', function () {
+        assert.strictEqual(connectionPool.close.callCount, 0);
+      });
+      it('Should not log anything', function () {
+        Object.values(testLogger).forEach((logStub) => {
+          assert.strictEqual(logStub.callCount, 0);
+        });
+      });
+    });
+    context('When there is an existing connectionPool', function () {
+      beforeEach(async function () {
+        return wrapper.getConnection();
+      });
+      context('When the pool fails to release', function () {
+        beforeEach(function () {
+          connectionPool.close.rejects(testError);
+        });
+        it('Should call connectionPool.close', async function () {
+          try {
+            await wrapper.releasePool();
+            assert.fail();
+          } catch (error) {
+            assert.strictEqual(connectionPool.close.callCount, 1);
+          }
+        });
+        it('Should log an error message', async function () {
+          try {
+            await wrapper.releasePool();
+            assert.fail();
+          } catch (error) {
+            assert.strictEqual(testLogger.error.callCount, 1);
+            assert.strictEqual(
+              testLogger.error.args[0][0],
+              'Could not release pool'
+            );
+          }
+        });
+        it('Should rethrow the error', async function () {
+          try {
+            await wrapper.releasePool();
+            assert.fail();
+          } catch (error) {
+            assert.strictEqual(error, testError);
+          }
+        });
+      });
+      context('When the pool successfully releases', function () {
+        beforeEach(async function () {
+          connectionPool.close.resolves();
+          return wrapper.releasePool();
+        });
+        it('Should call connectionPool.close', function () {
+          assert.strictEqual(connectionPool.close.callCount, 1);
+        });
+        it('Should log a message about the connection pool releasing', function () {
+          assert.strictEqual(testLogger.info.callCount, 1);
+          assert.strictEqual(
+            testLogger.info.args[0][0],
+            'Connection pool released'
+          );
+        });
+      });
+    });
+  });
+  describe('query', function () {
+    let testError: Error;
+    let testQuery: string;
+    let testQueryWithoutVariables: string;
+    let testVariables: Record<string, string>;
+    beforeEach(function () {
+      testError = new Error('query failed');
+      testQueryWithoutVariables = 'SELECT * FROM USERS;';
+      testQuery = "SELECT * FROM USERS WHERE NAME=':name';";
+      testVariables = { name: 'foo' };
```

Comment:
Again, maybe these should all be in a test data file?

Reply:
Done as part of 6a7ade2